src/dune_engine/scheduler.ml

```diff
@@ -737,13 +737,12 @@ end = struct
     | true -> invalidation
     | false ->
       (* In this mode, we do not assume that all file system dependencies are
-         declared correctly and therefore conservatively require a rebuild. *)
-      (* The fact that the [events] list is non-empty justifies clearing the
+         declared correctly and therefore conservatively require a rebuild.
+
+         The fact that the [events] list is non-empty justifies clearing the
          caches. *)
       let (_ : _ Nonempty_list.t) = events in
-      (* Since [clear_caches] is not sufficient to guarantee invalidation, we
-         pay attention to [invalidation] too, for good measure *)
-      Memo.Invalidation.combine Memo.Invalidation.clear_caches invalidation
```

Comment:
@aalekseyev Perhaps, we could keep this just in case, since the semantics of `clear_caches` is still dodgy.

Reply:
As I said verbally, I think I prefer the simpler code. In the possibly-impossible corner case where `clear_caches` is broken, the `invalidation` generated from notifications will be broken too, so there's no point combining them.